syntax = "proto3";

message OrderRequest {
    string user_id = 1;
    string symbol = 2;
    string order_size = 3;
    string price = 4;
}

message OrderResponse {
    string status = 1;
}

message OrderStatusResponse {
    string order_id = 1;
    string user_id = 2;
    string symbol = 3;
    int32 amount = 4;
    string status = 5;
}

message OrderStatusAllRequest {}
message OrderStatusAllResponse {
    repeated OrderStatusResponse orders = 1;
}

service Order {
    rpc CreateOrder(OrderRequest) returns(OrderResponse) {};
    rpc GetAllOrders(OrderStatusAllRequest) returns(OrderStatusAllResponse) {};
}
